samples/sensor/magn_polling/sample.yaml

```diff
@@ -1,8 +1,18 @@
 sample:
-  name: BMC150 Magnetometer Sample
+  name: Magnetometer Polling Sample
+common:
+  harness: sensor
+  tags: sensors
+  filter: dt_alias_exists("magn0")
 tests:
-  sample.sensor.magn_polling:
-    harness: sensor
-    tags: sensors
-    depends_on: i2c
-    filter: dt_compat_enabled("bosch,bmc150_magn")
+  sample.sensor.magn_polling.i2c:
+    integration_platforms:
+      - frdm_k64f               # fxos8700
+      - thingy53_nrf5340_cpuapp # bmm150
+      - stm32f411e_disco        # lsm303agr_magn
+      - stm32f3_disco           # lsm303dlhc_magn
+  sample.sensor.magn_polling.spi:
+    extra_configs:
+      -  CONFIG_SPI=y
+    integration_platforms:
+      - sensortile_box          # lis2mdl
```

Comment:
I wonder if this can be handled in the board Kconfig.defconfig:

```
config SPI
    default y if XXX
```

Reply:
This would need to be done in each board on which we want to run this sample variant while proposed way allows to run the test with multiple boards.
So I think there pros and cons in both approaches.

Reply:
this technique is used e.g. for the display sample, I find it useful because board specifics are kept in board directory. but not a blocker because we have no rules for that.